apps/web/src/app/[orgShortcode]/settings/user/security/page.tsx

```diff
@@ -581,3 +620,180 @@ export default function Page() {
+                      key={org.publicId}
+                      publicId={org.publicId}
+                      avatarTimestamp={org.avatarTimestamp}
+                      name={org.name}
+                      shortcode={org.shortcode}
+                    />
+                  ))}
+                </>
+              )}
+              {accountData?.memberOrgs &&
+                accountData?.memberOrgs.length > 0 && (
+                  <>
+                    <span className="text-base-11 text-xs">
+                      {(accountData?.memberOrgs?.length ?? 0) > 1
+                        ? 'Leave these organizations:'
+                        : 'Leave this organization:'}
+                    </span>
+                    {accountData?.memberOrgs.map((org) => (
+                      <OrgObject
+                        key={org.publicId}
+                        publicId={org.publicId}
+                        avatarTimestamp={org.avatarTimestamp}
+                        name={org.name}
+                        shortcode={org.shortcode}
+                      />
+                    ))}
+                  </>
+                )}
+            </div>
+
+            <div className="flex w-96 flex-col gap-4">
+              <span>
+                Please enter your username{' '}
+                <span className="bg-base-11 text-base-1 rounded-md px-1 py-0.5 font-mono text-sm">
+                  {accountData?.username}
+                </span>{' '}
+                to confirm your account deletion.
+              </span>
+              <Form {...form}>
+                <FormField
+                  control={form.control}
+                  name="username"
+                  render={({ field }) => (
+                    <FormItem>
+                      <FormControl>
+                        <Input
+                          label="Username"
+                          inputSize="lg"
+                          {...field}
+                        />
+                      </FormControl>
+                      <FormMessage />
+                    </FormItem>
+                  )}
+                />
+              </Form>
+            </div>
+
+            <Button
+              variant="destructive"
+              className="mt-3 w-fit gap-2"
+              disabled={!validated}
+              loading={deletingAccount}
+              onClick={async (e) => {
+                e.preventDefault();
+                if (!validated)
+                  return form.setError('username', {
+                    message: 'Must match your username exactly'
+                  });
+                await form.handleSubmit(async () => deleteAccountConfirm())(e);
+              }}>
+              <Trash
+                size={16}
+                weight="bold"
+              />
+              <span>Yes I want to delete my whole account</span>
+            </Button>
+          </div>
+        )}
+      </div>
+    </>
+  );
+}
+
+function OrgObject({
+  key,
```

Comment:
you don't need to pass key like that, its handled by react. Passing key as prop would result in bugs